### PR TITLE
fix upgrade catalog condition

### PR DESCRIPTION
**Pull request by gitlawr**
And quote command paras to properly handle empty values.
To solve https://github.com/rancher/rancher/issues/10708

### CODE REVIEW

File: provider/jenkins/jenkins_provider.go [hunk 1]
```diff
@@ -679,9 +679,9 @@ func commandBuilder(activity *model.Activity, step *model.Step) string {
 		readme = EscapeShell(activity, readme)
 		answers := EscapeShell(activity, step.Answers)
 
-		endpoint := "$CATTLE_URL"
-		accessKey := "$CATTLE_ACCESS_KEY"
-		envKey := "$CATTLE_SECRET_KEY"
+		var endpoint string
```
Comment (orangedeng) [inline provider/jenkins/jenkins_provider.go]: 这里不是endpoint永远被初始化为空字符串吗？下面的endpoint != ""不是永远为false么

File: provider/jenkins/jenkins_provider.go [hunk 2]
```diff
@@ -679,9 +679,9 @@ func commandBuilder(activity *model.Activity, step *model.Step) string {
...
+		var accessKey string
+		var envKey string
 		var err error
 		if endpoint != "" {
```
Comment (orangedeng) [inline provider/jenkins/jenkins_provider.go]: here

File: provider/jenkins/jenkins_provider.go (hunk 1 above)
Comment (gitlawr) [reply provider/jenkins/jenkins_provider.go]: Right, it should be `step.Endpoint`.

Comment (gitlawr) [reply provider/jenkins/jenkins_provider.go]: Updated

**Comment by orangedeng** [comment]
LGTM

**Pull request merged**